pkg/resources/diff_suppressions_test.go

```diff
@@ -80,3 +80,73 @@ func Test_IgnoreAfterCreation(t *testing.T) {
 		assert.True(t, result)
 	})
 }
+
+func Test_NormalizeAndCompareIdentifiersSet(t *testing.T) {
+	rawDataWithValues := func(values []any) *schema.ResourceData {
+		return schema.TestResourceDataRaw(t, map[string]*schema.Schema{
+			"value": {
+				Required: true,
+				Type:     schema.TypeSet,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+		}, map[string]any{
+			"value": values,
+		})
+	}
+	emptyResourceData := rawDataWithValues([]any{})
+
+	t.Run("validation: size key", func(t *testing.T) {
+		assert.False(t, resources.NormalizeAndCompareIdentifiersInSet("value")("value.#", "1", "2", emptyResourceData))
+	})
+
+	t.Run("validation: case mismatch", func(t *testing.T) {
+		resourceData := rawDataWithValues([]any{"SCHEMA.OBJECT.IDENTIFIER"})
+		assert.False(t, resources.NormalizeAndCompareIdentifiersInSet("value")("value.doesnt_matter", "schema.object.identifier", "", resourceData))
+		// TODO: Cannot be tested with schema.TestResourceDataRaw because it doesn't populate raw state which is used in the cases below
```

Comment:
We already have an issue for the similar problem with TestResourceDataRaw problem: SNOW-1511594

Reply:
Added to the TODO comment